gossip/src/push_active_set.rs

```diff
@@ -110,14 +111,20 @@ impl PushActiveSetEntry {
 
     fn get_nodes<'a>(
         &'a self,
-        origin: &'a Pubkey,
+        pubkey: &'a Pubkey, // This node.
+        origin: &'a Pubkey, // CRDS value owner.
         // If true forces gossip push even if the node has pruned the origin.
         mut should_force_push: impl FnMut(&Pubkey) -> bool + 'a,
     ) -> impl Iterator<Item = &Pubkey> + 'a {
+        let pubkey_eq_origin = pubkey == origin;
         self.0
             .iter()
             .filter(move |(node, bloom_filter)| {
-                !bloom_filter.contains(origin) || should_force_push(node)
+                // Bloom filter can return false positive for origin == pubkey
+                // but a node should always be able to push its own values.
+                !bloom_filter.contains(origin)
+                    || (pubkey_eq_origin && &pubkey != node)
```

Comment:
not sure I understand this check:
```
&pubkey != node
```
When would a node have its own pubkey as the value in the `PushActiveSetEntry` index map?
Is it possible for a node to add itself to its `PushActiveSet`? Even if it is, wouldn't we not want to send the message to ourselves if `&pubkey == node`?

Reply:
> When would a node have its own pubkey as the value in the PushActiveSetEntry index map?

The code here does not have any guarantees that `pubkey != node`.
So we need to check here not to send our own messages back to ourselves.

> Is it possible for a node to add itself to its PushActiveSet?

We exclude that here:
https://github.com/anza-xyz/agave/blob/ce158213f/gossip/src/crds_gossip.rs#L352
but nonetheless the code here does not need to rely on that presumption.

> Even if it is, wouldn't we not want to send the message to ourselves if &pubkey == node?

not sure I understand this.
`&& &pubkey != node` prevents sending messages to ourselves.

Reply:
ok that makes sense. 
and ya you're right I screwed up what we wanted to return in the filter.